### PR TITLE
Floor function addition

### PR DESCRIPTION
**Pull request by saatviks**
Added Floor function following the [Postgres standard](https://www.postgresql.org/docs/9.5/static/functions-math.html). Added 2 unit tests - one to check working of the Floor function internally and one with SQL.

### CODE REVIEW

File: test/sql/decimal_functions_sql_test.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// distinct_sql_test.cpp
```
Comment (apavlo) [inline test/sql/decimal_functions_sql_test.cpp]: Run the formatter script to update header:
```
$ ./script/formatting/formatter.py --add-header  test/sql/decimal_functions_sql_test.cpp
```

File: test/sql/decimal_functions_sql_test.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      TestingSQLUtil::ExecuteSQLQuery(os.str());
+    }
+    EXPECT_EQ(i, numEntries);
+
+    // Fetch values from the table
+    std::vector<StatementResult> result;
+    std::vector<FieldInfo> tuple_descriptor;
+    std::string error_message;
+    int rows_affected;
+    std::ostringstream os;
```
Comment (apavlo) [inline test/sql/decimal_functions_sql_test.cpp]: Why not just use `std::string` here?

File: test/sql/decimal_functions_sql_test.cpp (hunk 1 above)
Comment (saatviks) [reply test/sql/decimal_functions_sql_test.cpp]: Addressed in latest commit

File: test/sql/decimal_functions_sql_test.cpp (hunk 2 above)
Comment (saatviks) [reply test/sql/decimal_functions_sql_test.cpp]: Addressed in latest commit

**Pull request merged**